svgdigitizer/electrochemistry/cv.py

```diff
@@ -814,28 +981,39 @@ def metadata(self):
             ...   </g>
             ...   <text x="-200" y="330">scan rate: 50 V/s</text>
             ...   <text x="-400" y="430">comment: noisy data</text>
-            ...   <text x="-400" y="430">linked: SXRD, SHG</text>
+            ...   <text x="-400" y="530">linked: SXRD, SHG</text>
+            ...   <text x="-200" y="630">Figure: 2b</text>
+            ...   <text x="-200" y="730">tags: BCV, HER, OER</text>
             ... </svg>'''))
             >>> cv = CV(SVGPlot(svg))
             >>> cv.metadata  # doctest: +NORMALIZE_WHITESPACE
-            {'figure description': {'version': 1,
-              'type': 'digitized',
-              'simultaneous measurements': ['SXRD', 'SHG'],
-              'measurement type': 'CV',
-              'scan rate': {'value': 50.0, 'unit': 'V / s'},
-              'axes': {'E': {'unit': 'mV', 'reference': 'RHE', 'orientation': 'x'},
-               'j': {'unit': 'uA / cm2', 'orientation': 'y'},
-               't': {'unit': 's'}},
-              'comment': 'noisy data'},
-             'data description': {'version': 1,
-              'type': 'digitized',
-              'measurement type': 'CV',
-              'axes': {'E': {'unit': 'V', 'reference': 'RHE'},
-               'j': {'unit': 'A / m2'},
-               't': {'unit': 's'}}}}
+            {'experimental': {'tags': ['BCV', 'HER', 'OER']},
+             'source': {'figure': '2b', 'curve': '0'},
+             'figure description': {'version': 1,
+             'type': 'digitized',
+             'simultaneous measurements': ['SXRD', 'SHG'],
+             'measurement type': 'CV',
+             'scan rate': {'value': 50.0, 'unit': 'V / s'},
+             'fields': [{'name': 'E', 'unit': 'mV',
+                        'reference': 'RHE', 'orientation': 'x'},
+                        {'name': 'j', 'unit': 'uA / cm2',
+                        'orientation': 'y'}],
+                        'comment': 'noisy data'},
+             'data description': {'version': 1, 'type': 'digitized',
+                                  'measurement type': 'CV',
+                                  'fields': [{'name': 't', 'unit': 's'},
+                                  {'name': 'E', 'unit': 'V', 'reference': 'RHE'},
+                                  {'name': 'j', 'unit': 'A / m2'}]}}
 
         """
         metadata = self._metadata.copy()
+        # Add experimental to metadata
+        metadata.setdefault("experimental", {})
+        metadata["experimental"]["tags"] = self.tags
```

Comment:
This will override `tags` from the yaml even if `self.tags==[]`. Is that intentional now?

Reply:
That was more or less what we decided on in the past for other entries. By overriding the keys in the YAML it is clear that only the keys in the SVG file matter.

Reply:
Sure. But you're overriding them even if there are none in the SVG.

Reply:
ok, so you mean that the key should not exist when it is empty?

Reply:
Yes, I think that the tags should not exist when they are missing from both the YAML and the SVG.

But when tags are given in the YAML but *not* in the SVG, then they should certainly not be empty.

Reply:
Let me try to propose a better implementation here.

Reply:
I think we should simply overwrite the tags in the yaml. At the moment tags do not yet exist. I currently add them to the existing SVG files in the website repo.